ui/ui-components/pages/model/[modelId]/source/[sourceId]/edit.tsx

```diff
@@ -83,10 +83,9 @@ const Page: NextPage<Props & InjectedProps> = ({
   );
   const isPrimarySource = useMemo(
     () =>
-      (totalSources === 1 && source.state !== "ready") ||
       properties.filter(
         ({ isPrimaryKey, sourceId }) => isPrimaryKey && sourceId === source.id
-      ).length > 0,
+      ).length > 0 || totalSources === 1,
```

Comment:
The logic can be switched to `totalSources == 1 || properties.filter...` because checking if a value is 1 is a less expensive operation than filter